lib/types/subnet_info.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Types.SubnetInfo do
+  @moduledoc """
+  Struct to hold subnet attestations for easier db storing:
+  - data: An attestation data.
+  - attestations: List of all the collected Attestations.
+  """
+
+  defstruct [:data, :attestations]
+
+  @type t :: %__MODULE__{
+          data: Types.AttestationData.t(),
+          attestations: list(Types.Attestation.t())
+        }
+
+  @spec new_subnet_with_attestation(Types.Attestation.t()) :: Types.SubnetInfo.t()
+  def new_subnet_with_attestation(%Types.Attestation{data: data} = attestation),
```

Comment:
Let's document this function and explain that the attestation will be used to filter other attestations added, and that it's typically the attestation that a validator built before starting to collect attestations.